src/insights_client/metrics.py

```diff
@@ -9,10 +9,13 @@
 
 
 def _proxy_settings(rhsm_config):
-    hostname = rhsm_config.get("server", "proxy_hostname").strip()
-    port = rhsm_config.get("server", "proxy_port").strip()
-    user = rhsm_config.get("server", "proxy_user").strip()
-    password = rhsm_config.get("server", "proxy_password").strip()
+    try:
+        hostname = rhsm_config.get("server", "proxy_hostname").strip()
+        port = rhsm_config.get("server", "proxy_port").strip()
+        user = rhsm_config.get("server", "proxy_user").strip()
+        password = rhsm_config.get("server", "proxy_password").strip()
+    except (configparser.NoSectionError, configparser.NoOptionError):
+        return None
 
     if not hostname:
         return None
```

Comment:
This can be also moved below the _hostname_ variable assignment to shortcut when we already know that we don’t need to read the rest. But that’s not important.

```suggestion
```